lighthouse-core/test/lib/network-request-test.js

```diff
@@ -0,0 +1,243 @@
+    }
+
+    it('does nothing if not Lightrider', function() {
+      const req = getRequest();
+      global.isLightrider = false;
+
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+      req._updateTimingsForLightrider();
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+    });
+
+    it('does nothing if no TotalTime', function() {
+      const req = getRequest();
+      global.isLightrider = true;
+      req.responseHeaders = req.responseHeaders.filter(item => item.name
+        !== NetworkRequest.HEADER_TOTAL);
+
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+      req._updateTimingsForLightrider();
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+    });
+
+    it('does nothing if Header timings don\'t add up', function() {
+      const req = getRequest();
+      global.isLightrider = true;
+      req.responseHeaders = [{name: NetworkRequest.HEADER_TOTAL, value: 10000},
+        {name: NetworkRequest.HEADER_TCP, value: 5001},
+        {name: NetworkRequest.HEADER_REQ, value: 2500},
+        {name: NetworkRequest.HEADER_SSL, value: 1000},
+        {name: NetworkRequest.HEADER_RES, value: 2500}];
+
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+      req._updateTimingsForLightrider();
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+    });
+
+    it('does nothing if timing is not initialized', function() {
+      const req = getRequest();
+      global.isLightrider = true;
+      req.timing = undefined;
+
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+      req._updateTimingsForLightrider();
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+    });
+
+    it.skip('Handles negative timing data', function() {
+      const req = getRequest();
+      global.isLightrider = true;
+      req.responseHeaders = [{name: NetworkRequest.HEADER_TOTAL, value: 10000},
+        {name: NetworkRequest.HEADER_TCP, value: -1},
+        {name: NetworkRequest.HEADER_REQ, value: -1},
+        {name: NetworkRequest.HEADER_SSL, value: -1},
+        {name: NetworkRequest.HEADER_RES, value: 10000}];
+
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 2);
+      assert.equal(req.responseReceivedTime, 1);
+      req._updateTimingsForLightrider();
+      assert.equal(req.startTime, 0);
+      assert.equal(req.endTime, 10);
+      assert.equal(req.responseReceivedTime, 0);
+    });
+
+    it.skip('updates fetch stats', function() {
```

Comment:
these skip() tests should exeunt to another PR with the other bit.

Reply:
![image](https://user-images.githubusercontent.com/4071474/55662426-cd8a1d80-57c7-11e9-9310-879a6d7f7384.png)

Reply:
I need to read more books so I can understand these PR reviews...